application/src/main/java/org/togetherjava/tjbot/commands/basic/RoleSelectCommand.java

```diff
@@ -0,0 +1,319 @@
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
+
+/**
+ * Implements the {@code roleSelect} command.
+ *
+ * <p>
+ * Allows users to select their roles without using reactions, instead it uses selection menus where
+ * you can select multiple roles. <br />
+ * Note: the bot can only use roles with a position below its highest one
+ */
+public class RoleSelectCommand extends SlashCommandAdapter {
+
+    private static final Logger logger = LoggerFactory.getLogger(RoleSelectCommand.class);
+
+    private static final String ALL_OPTION = "all";
+    private static final String CHOOSE_OPTION = "choose";
+
+    private static final String TITLE_OPTION = "title";
+    private static final String DESCRIPTION_OPTION = "description";
+
+    private static final Color AMBIENT_COLOR = new Color(24, 221, 136, 255);
+
+    private static final List<OptionData> messageOptions = List.of(
+            new OptionData(OptionType.STRING, TITLE_OPTION, "The title for the message", false),
+            new OptionData(OptionType.STRING, DESCRIPTION_OPTION, "A description for the message",
+                    false));
+
+
+    /**
+     * Construct an instance
+     */
+    public RoleSelectCommand() {
+        super("role-select", "Sends a message where users can select their roles",
+                SlashCommandVisibility.GUILD);
+
+        SubcommandData allRoles =
+                new SubcommandData(ALL_OPTION, "Lists all the rolls in the server for users")
+                    .addOptions(messageOptions);
+
+        SubcommandData selectRoles =
+                new SubcommandData(CHOOSE_OPTION, "Choose the roles for users to select")
+                    .addOptions(messageOptions);
+
+        getData().addSubcommands(allRoles, selectRoles);
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull final SlashCommandEvent event) {
+        Member member = Objects.requireNonNull(event.getMember(), "Member is null");
+        if (!member.hasPermission(Permission.MANAGE_ROLES)) {
+            event.reply("You dont have the right permissions to use this command")
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+
+        Member selfMember = Objects.requireNonNull(event.getGuild()).getSelfMember();
+        if (!selfMember.hasPermission(Permission.MANAGE_ROLES)) {
+            event.reply("The bot needs the manage role permissions").setEphemeral(true).queue();
+            logger.warn("The bot needs the manage role permissions");
+            return;
+        }
+
+        SelectionMenu.Builder menu = SelectionMenu.create(generateComponentId(member.getId()));
```

Comment:
you have to choose `Lifetime.PERMANENT` in `generateComponentId`, otherwise the component id might get stale after a few weeks/months. (see the overload of the generate method)